packages/KInlineEdit/KInlineEdit.vue

```diff
@@ -0,0 +1,122 @@
+<template>
+  <div
+    id="editable-wrapper"
+    class="k-inline-edit">
+    <input
+      v-if="isEditing"
+      ref="input"
+      :style="styles"
+      v-model="inputText"
+      class="k-input"
+      @blur="handleSave"
+      @keyup.enter="handleSave">
+    <div
+      id="element-content-wrapper"
+      @click="handleClick">
+      <slot v-if="!isEditing" />
+    </div>
+  </div>
+</template>
+
+<script>
+// Styles we want copied from the element
+const STYLES = {
+  fontSize: 'font-size',
+  fontWeight: 'font-weight',
+  fontFamily: 'font-family',
+  color: 'color',
+  margin: 'margin',
+  padding: 'padding'
+}
+
+export default {
+  name: 'KInlineEdit',
+  data () {
+    return {
+      isEditing: false,
+      inputText: '',
+      styles: {}
+    }
+  },
+
+  mounted () {
+    if (this.$slots.default[0].tag === undefined) {
+      console.warn(`KInlineEdit expects slotted HTML tag.
+
+Example usage:
+
+<KInlineEdit>
+  <p>Some text</p>
+</KInlineEdit>
+      `)
+    }
+  },
+
+  methods: {
+    async handleClick (e) {
+      // If clicking the slot wrapper lets exit out as to not
+      // copy its styles to the input
+      if (e.target.id === 'element-content-wrapper') return
+
+      // Get current STYLES off of the element
+      this.styles = this.getStyles(e.target)
+      this.inputText = e.target.innerText
+      this.isEditing = true
+
+      // Wait for vue to update styles & text
+      await this.$nextTick()
+      this.$refs.input.focus()
+    },
+
+    handleSave (e) {
+      this.isEditing = false
+      this.$emit('changed', this.inputText)
+    },
+
+    getStyles (element) {
+      const elementStyles = getComputedStyle(element)
+
+      return Object.keys(STYLES).reduce((acc, cur) => {
+        acc[cur] = elementStyles.getPropertyValue(STYLES[cur])
+
+        return acc
+      }, {})
+    }
+  }
+}
+</script>
+
+<style lang="scss" scoped>
+@import '~@kongponents/styles/_variables.scss';
+@import '~@kongponents/styles/forms/_inputs.scss';
+
+.k-inline-edit {
+  --padding: var(--spacing-xxs) var(--spacing-xs);
+  box-sizing: border-box;
+  > div {
+    display: inline-flex;
+    cursor: text;
+    > * {
+      border: 1px solid transparent;
+      border-radius: 3px;
+      padding: var(--padding);
+      margin-left: calc(-1 * var(--spacing-xs)); // align the left side of content
+      line-height: 1.20;
+      overflow: hidden;
+      transition: background-color 200ms ease;
+    }
+    &:hover > * {
+      background-color: var(--grey-92);
+    }
+  }
+  .k-input {
+    display: inline-flex;
+    width: 100%;
```

Comment:
this makes the k-input 100% once you click it, but the hoverable-area is still limited to just the target element width itself.

Reply:
ah I thought you meant just the input. I'll move up to the wrapper to so it inherits from the overall parent.

Reply:
made everything fullwidth